lib/bureaucrat/swagger_slate_markdown_writer.ex

```diff
@@ -0,0 +1,328 @@
+    write_models(file, swagger)
+
+    records
+      |> tag_records(swagger)
+      |> group_records()
+      |> Enum.each(fn {tag, records_by_operation_id} ->
+           write_operations_for_tag(file, tag, records_by_operation_id, swagger)
+         end)
+  end
+
+  @doc """
+  Writes the document title and api summary description.
+
+  This corresponds to the info section of the swagger document.
+  """
+  def write_overview(file, swagger) do
+    info = swagger["info"]
+    file
+    |> puts("""
+    ---
+    title: #{info["title"]}
+
+    search: true
+    ---
+
+    # #{info["title"]}
+
+    #{info["description"]}
+    """)
+  end
+
+  @doc """
+  Writes the authentication details to the given file.
+
+  This corresponds to the securityDefinitions section of the swagger document.
+  """
+  def write_authentication(file, swagger) do
+    file
+    |> puts("# Authentication\n")
+
+    # TODO: Document token based security
+    Enum.each swagger["security"], fn {name, _details} ->
+       definition = swagger["securityDefinitions"][name]
+       file
+       |> puts("## #{definition["type"]}\n")
+       |> puts("#{definition["description"]}\n")
+     end
+  end
+
+  @doc """
+  Writes the API request/response model schemas to the given file.
+
+  This corresponds to the definitions section of the swagger document.
+  Each top level definition will be written as a table.
+  Nested objects are flattened out to reduce the number of tables being produced.
+  """
+  def write_models(file, swagger) do
+    puts(file, "# Models\n")
+    Enum.each swagger["definitions"], fn definition ->
+      write_model(file, definition)
+    end
+    file
+  end
+
+  @doc """
+  Writes a single API model schema to the given file.
+
+  Most of the work is delegated to the write_model_properties/3 recurive function.
+  The example json is output before the table just so slate will align them.
+  """
+  def write_model(file, {name, model_schema}) do
+    {:ok, json} = Poison.encode(model_schema["example"], pretty: true)
+
+    file
+    |> puts("## #{name}\n")
+    |> puts("#{model_schema["description"]}")
+    |> puts("\n```json")
+    |> puts(json)
+    |> puts("```\n")
+    |> puts("|Property|Description|Type|Required|")
+    |> puts("|--------|-----------|----|--------|")
+    |> write_model_properties(model_schema)
+  end
+
+  @doc """
+  Writes the fields of the given model to file.
+
+  prefix is output before each property name to enable nested objects to be flattened.
+  """
+  def write_model_properties(file, model_schema, prefix \\ "") do
+    Enum.each model_schema["properties"], fn {property, property_details} ->
+      required? = property in model_schema["required"]
+      type = property_details["type"]
+      case type do
+        "object" ->
+          #TODO: handle object with schema reference
+          write_model_properties(file, property_details, prefix <> "#{property}.")
+        _ ->
+          if type == "array" do
```

Comment:
maybe better `write_model_properties` pass last argument like `object `, `array `
Then this function can be more easy.